### PR TITLE
[WIP] Add specs for work_file_required config, refs #1715 -- TESTING, DO NOT MERGE

### PR DESCRIPTION
**Pull request by hackmastera**

### CODE REVIEW

**Comment by hackmastera** [comment]
expect these to fail but they were passing for me locally.

File: spec/javascripts/uploaded_files_spec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        let element = fixture.find('form')
+        target = new control.UploadFiles(element);
+        expect(target.hasFileRequirement).toBe(true);
+      });
+    });
+
+    describe("without required file element", function() {
+      it("returns false", function() {
+        let fixture = setFixtures('<form><ul class="requirements"><li>Not files</li></ul></form>');
+        let = fixture.find('form')
```
Comment (mjgiarlo) [inline spec/javascripts/uploaded_files_spec.js]: missing `element` on this line, methinks

**Comment by mjgiarlo** [comment]
@jcoyne @awead @cam156 Any thoughts on the above, folks? This is related to @hackmastera's PR #2406.

**Comment by hackmastera** [comment]
Thanks, all, and it helped to find the jasmine server. With this update the jasmine server shows the specs failing as expected, but rspec shows all jasmine tests passing.

screen shot 2016-08-10 at 10 26 46 am

**Comment by hackmastera** [comment]
@jcoyne you're right it isn't running them all in rspec (20 vs. 22 specs run). Is there some way to add them? It is finding all 7 files.

**Comment by cam156** [comment]
@hackmastera Unfortunately if there is syntax error it just does not run in rspec.  That is part of the reason I output the number.   When I have tried to find an error just using rspec in the past you can specify something you know will fail so you can see values like
```
expect(target).toBe('abc');
```

File: spec/javascripts/save_work_spec.js [hunk 2]
```diff
@@ -99,6 +99,7 @@ describe("SaveWorkControl", function() {
         target = buildTarget(form_id)
         target.uploads = {
           hasFiles: true
```
Comment (mjgiarlo) [inline spec/javascripts/save_work_spec.js]: Does this line need a comma on the end?

File: spec/javascripts/save_work_spec.js [hunk 3]
```diff
@@ -113,6 +114,7 @@ describe("SaveWorkControl", function() {
         target = buildTarget(form_id)
         target.uploads = {
           hasFiles: false
```
Comment (mjgiarlo) [inline spec/javascripts/save_work_spec.js]: Does this line need a comma on the end?

**Comment by mjgiarlo** [comment]
I commented on a couple of lines that may have syntax errors.

I wonder if we might want to turn on integration with a service like HoundCI that can do JavaScript and SCSS linting for us to find errors like these?

File: spec/javascripts/uploaded_files_spec.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+describe("UploadedFiles", function() {
+  var control = require('sufia/save_work/uploaded_files');
+
+  describe("hasFileRequirement", function() {
+    describe("with required file element", function() {
+      it("returns true", function() {
+        let fixture = setFixtures('<form><ul class="requirements"><li class="incomplete" id="required-files">Add files</li></ul></form>');
+        let element = fixture.find('form')
```
Comment (mjgiarlo) [inline spec/javascripts/uploaded_files_spec.js]: Does this line need a semi-colon on the end?

File: spec/javascripts/uploaded_files_spec.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        target = new control.UploadedFiles(element);
+        expect(target.hasFileRequirement).toBe(true);
+      });
+    });
+
+    describe("without required file element", function() {
+      it("returns false", function() {
+        let fixture = setFixtures('<form><ul class="requirements"><li>Not files</li></ul></form>');
+        let element = fixture.find('form')
```
Comment (mjgiarlo) [inline spec/javascripts/uploaded_files_spec.js]: Semi-colon need at end of line?

File: spec/javascripts/save_work_spec.js (hunk 2 above)
Comment (hackmastera) [inline spec/javascripts/save_work_spec.js]: that test is running. but i should probably still add a comma.

**Comment by hackmastera** [comment]
@cam156 thanks for that info. @mjgiarlo this is now failing as expected with all tests running (31); i'll close and delete the branch.

**Comment by mjgiarlo** [comment]
:clap: @hackmastera

**Pull request closed without merging**